soci/soci_index.go

```diff
@@ -94,6 +112,60 @@ type IndexDescriptorInfo struct {
 	CreatedAt time.Time
 }
 
+// DecodeIndex deserializes a JSON blob in an io.Reader
+// into a SOCI index. The blob can be either an OCI 1.1 Artifact
+// or an OCI 1.0 Manifest
+func DecodeIndex(r io.Reader, index *Index) error {
+	b, err := io.ReadAll(r)
+	if err != nil {
+		return err
+	}
+	return UnmarshalIndex(b, index)
+}
+
+// UnmarshalIndex deserializes a JSON blob in a byte array
+// into a SOCI index. The blob can be either an OCI 1.1 Artifact
+// or an OCI 1.0 Manifest
+func UnmarshalIndex(b []byte, index *Index) error {
+	err := json.Unmarshal(b, index)
+	if err == nil && index.MediaType == ocispec.MediaTypeArtifactManifest {
```

Comment:
maybe add a log statement (debug?) that this is a 1.1 artifact?

Reply:
It's awkward to put logs here - we would need to pass a context through and this feels like a piece of the core library that other projects could build on.